nautobot/extras/jobs.py

```diff
@@ -828,6 +835,35 @@ def __init__(self, min_prefix_length=None, max_prefix_length=None, *args, **kwar
             self.field_attrs["validators"].append(MaxPrefixLengthValidator(max_prefix_length))
 
 
+class JobHookReceiver(Job):
+    """
+    Base class for job hook receivers. Job hook receivers are jobs that are initiated
+    from object changes and are not intended to be run from the UI or API like standard jobs.
+    """
+
+    object_change = ObjectVar(model=ObjectChange)
+
+    def run(self, data, commit):
+        """JobHookReceiver subclasses generally shouldn't need to override this method."""
+        object_change = data["object_change"]
```

Comment:
Should we check/assert that `commit` is always `True` for `JobHookReceiver.run()`?

Reply:
`raise ValidationError` if it's not `True`?